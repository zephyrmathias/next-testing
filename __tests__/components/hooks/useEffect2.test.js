import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import Example from '../../../components/hooks/useEffect2'

describe('onClick', () => {
  it('should increase count number when button is clicked', () => {
    const { container } = render(<Example />)
    const button = container.firstChild
    expect(button.textContent).toBe('0')

    // basically, this test also covers useEffect as componentDidUpdate
    fireEvent.click(button)
    expect(button.textContent).toBe('1')
  })
})

describe('diff snapshot', () => {
  it('should increase count number when button is clicked', () => {
    const { asFragment, container } = render(<Example />)
    const firstRender = asFragment()
    const button = container.firstChild

    fireEvent.click(button)

    expect(firstRender).toMatchDiffSnapshot(asFragment())
  })
})

