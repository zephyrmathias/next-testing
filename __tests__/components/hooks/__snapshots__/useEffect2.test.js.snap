// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`diff snapshot should increase count number when button is clicked 1`] = `
Snapshot Diff:
- First value
+ Second value

  <DocumentFragment>
    <button>
-     0
+     1
    </button>
  </DocumentFragment>
`;
